LMChatCore_iOS/LMChatCore_iOS/Source/Module/LMAIChatBotChat/LMAIChatBotChatViewModel.swift

```diff
@@ -0,0 +1,171 @@
+            delegate?.didFailInitialization(with: "Invalid chatbot UUID")
+            return
+        }
+        
+        let request = CheckDMStatusRequest.builder()
+            .requestFrom("member_profile")
+            .uuid(chatbotUUID)
+            .build()
+        
+        LMChatClient.shared.checkDMStatus(request: request) { [weak self] response in
+            guard let self = self else { return }
+            print("dmResponse is \(response.data)")
+            
+            // Check if DM is enabled
+            guard let showDM = response.data?.showDM, showDM else {
+                self.delegate?.didFailInitialization(with: "Direct messaging is not enabled")
+                return
+            }
+            
+            // Check if we have a CTA URL with chatroom ID
+            if let cta = response.data?.cta,
+               let ctaURL = URL(string: cta),
+               let components = URLComponents(url: ctaURL, resolvingAgainstBaseURL: false),
+               let chatroomId = components.queryItems?.first(where: { $0.name == "chatroom_id" })?.value {
+                
+                // Chatroom exists, save ID and proceed
+                self.saveAndNavigateToChatroom(chatroomId)
+            } else {
+                // No existing chatroom, create new one
+                self.createDMChatroom(with: chatbotUUID)
+            }
+        }
+    }
+    
+    /// Creates a new DM chatroom with the chatbot
+    private func createDMChatroom(with chatbotUUID: String) {
+        let request = CreateDMChatroomRequest.builder()
+            .uuid(chatbotUUID)
+            .build()
+        
+        LMChatClient.shared.createDMChatroom(request: request) { [weak self] response in
+            guard let self = self else { return }
+            
+            if let chatroomId = response.data?.chatroomData?.communityId {
+                self.saveAndNavigateToChatroom(chatroomId)
+            } else {
+                self.delegate?.didFailInitialization(with: "Failed to create chatroom")
+            }
+        }
+    }
+    
+    /// Saves the chatroom ID and navigates to the chatroom screen
+    private func saveAndNavigateToChatroom(_ chatroomId: String) {
+        // Save chatroom ID to local prefs
+        LMSharedPreferences.setString(chatroomId, forKey: "chatroomIdWithAIChatbot")
+        
+        // Navigate to chatroom first
+        DispatchQueue.main.async { [weak self] in
+            guard let viewController = self?.delegate as? LMViewController else { return }
+            
+            // Navigate to the chatroom screen
+            NavigationScreen.shared.perform(
+                .chatroom(chatroomId: chatroomId, conversationID: nil),
+                from: viewController,
+                params: nil
+            )
+            
+            // Then notify completion which will trigger dismissal
+            self?.delegate?.didCompleteInitialization()
```

Comment:
call didCompleteInitialisation first so that the loading screen will get dismissed, then call navigation code. Also please check other screen for navigation pattern. I don't think we do navigation from view models. This code should be in ViewController.

```suggestion
            self?.delegate?.didCompleteInitialization(chatroomId: chatroomId, conversationID: nil)
```

Modify `didCompleteInitialization` to accept chatroom id so that we can perform navigation from ViewController only.